edi_xml_oca/components/xml_handler.py

```diff
@@ -65,15 +68,13 @@ def validate(self, xml_content, raise_on_fail=False):
             * None if validation is ok
             * error string if `raise_on_fail` is False
         """
+        xsd_name = self.schema_path
+        xml_content = (
+            xml_content.encode("utf-8") if isinstance(xml_content, str) else xml_content
+        )
         try:
-            return self.schema.validate(xml_content)
-        except self._validate_swallable_exceptions() as err:
+            validate_xml_from_attachment(self.env, xml_content, xsd_name)
```

Comment:
Are you sure this method is doing what we want? Have you checked it's implementation?

Reply:
This public function validates the XML content with an XSD attachment; making a call to the private function `_check_with_xsd` that you pointed [here](https://github.com/OCA/edi-framework/issues/127#issuecomment-2531579658). Actually `validate_xml_from_attachment` acts as a wrapper to validate XML content against an XSD schema while handling exceptions gracefully.

The method implementation can be seen [here](https://github.com/odoo/odoo/blob/834e8a8789d1d451f62ae1e841845baf7d3931a4/odoo/tools/xml_utils.py#L315) 

Unless I'm missing something, yes I think the function does what we want :thinking: